skbio/io/format/sample_metadata.py

```diff
@@ -0,0 +1,55 @@
+from skbio.io import create_format
+from skbio.metadata._metadata import SampleMetadata, MetadataColumn
+from skbio.metadata.io import MetadataFileError, MetadataReader
+from skbio.metadata.base import (
+    SUPPORTED_COLUMN_TYPES,
+    FORMATTED_ID_HEADERS,
+    is_id_header,
+)
+from skbio.metadata._util import find_duplicates
+from skbio.metadata.missing import (
+    DEFAULT_MISSING,
+    BUILTIN_MISSING,
+    series_encode_missing,
+)
+
+import csv
+import itertools
+import pandas as pd
+
+
+sample_metadata = create_format("sample_metadata")
+
+
+@sample_metadata.sniffer()
+def _sample_metadata_sniffer(fh):
+    # Strategy:
+    # Check if first word is in the file is in the list
+    # of allowed metadata words
+    try:
+        tsv_reader = csv.reader(fh, dialect="excel-tab", strict=True)
```

Comment:
`header = open(fh, 'r').readline()`

Reply:
@mortonjt I am not sure about the context, but I guess `header = open(fh, 'r').readline()` will leave the file handle unclosed. The standard `with` way can avoid this issue.